core/templates/domain/statistics/playthrough-backend-api.service.ts

```diff
@@ -0,0 +1,75 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Backend API service for playthrough.
+ */
+
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+
+import { Playthrough } from
+  'domain/statistics/PlaythroughObjectFactory';
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service';
+
+interface IStorePlaythroughBackendResponse {
+  'playthrough_stored': boolean;
+  'playthrough_id': string;
+}
+
+export class StorePlaythroughResponse {
+  playthroughStored: boolean;
+  playthroughId: string;
+
+  constructor(playthroughStored: boolean, playthroughId: string) {
+    this.playthroughId = playthroughId;
+    this.playthroughStored = playthroughStored;
+  }
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class PlaythroughBackendApiService {
+  STORE_PLAYTHROUGH_URL: string = (
+    '/explorehandler/store_playthrough/<exploration_id>');
+
+  constructor(
+    private http: HttpClient,
+    private urlInterpolationService: UrlInterpolationService) {}
+
+  storePlaythrough(
+      playthrough: Playthrough, issueSchemaVersion: number,
+      playthroughId: string): Promise<StorePlaythroughResponse> {
+    let playthroughUrl = this.urlInterpolationService.interpolateUrl(
+      this.STORE_PLAYTHROUGH_URL, {
+        exploration_id: playthrough.expId
+      });
+
+    return this.http.post(playthroughUrl, {
```

Comment:
Seems my other comment was lost, but please add type information to `post` here:

```typescript
return this.http.post<IStorePlaythroughBackendResponse>(
  playthroughUrl, {
    ...
  }).toPromise().then(response => {
    return new StorePlaythroughResponse(...);
  });
```

Reply:
I think doing `response: IStorePlaythroughBackendResponse` is same as that. Isn't it?

Reply:
Yes, but the type safety is stronger when providing it to `post`. Also, Angular's entire goal by introducing the generic was to explicitly avoid needing to do `response: T`. Please change here and elsewhere.

Reply:
Oh, ok. I think then it should be a global change in all the backend-api-services?

Reply:
Yes, I thought it already was since I recall seeing it in the codebase already: https://github.com/oppia/oppia/blob/b95f0e978ad853f086ef89723bc51cc00dfee77d/core/templates/domain/topics_and_skills_dashboard/topics-and-skills-dashboard-backend-api.service.ts#L73-L90.

Reply:
@brianrodri Yeah, it is but not in all places.

Reply:
So, should we create an issue for this to follow this convention everywhere?

Reply:
Yes. Please start in this PR and we can clean up the rest incrementally.

Reply:
Filed #9453

Reply:
done